akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -778,19 +778,21 @@ private[akka] class ShardRegion(
   }
 
   def receiveCommand(cmd: ShardRegionCommand): Unit = cmd match {
-    case Retry =>
+
+    case Retry if gracefulShutdownInProgress =>
```

Comment:
Guards will break the compiler completeness check so a `case Retry => if (gracefulShutdownInProgress) {} else {}` could be better for that reason.